client/db/src/lib.rs

```diff
@@ -544,7 +544,12 @@ pub struct BlockImportOperation<Block: BlockT> {
 
 impl<Block: BlockT> BlockImportOperation<Block> {
 	fn apply_offchain(&mut self, transaction: &mut Transaction<DbHash>) {
-		for (key, value_operation) in self.offchain_storage_updates.drain() {
+		for ((prefix, key), value_operation) in self.offchain_storage_updates.drain() {
+			let key: Vec<u8> = prefix
+				.into_iter()
+				.chain(sp_core::sp_std::iter::once(b'/'))
```

Comment:
Can you elaborate on this change? It feels weird to alter the prefix during `apply` only for a specific DB backend. Why is this not part of the `offchain_storage_updates` already?

Reply:
Valid point, but there is really no point to generalize how to glue prefix and key together at the upper layer and possibly aliasing badly chosen prefixes at the overlay level.
I'd rather push that down to the lower layers which actually deal with.
It's also way more convenient not having to dissect the combined key into parts again.

Reply:
> It's also way more convenient not having to dissect the combined key into parts again.

Why would that ever be needed? Users of on-chain and off-chain access should simply use the same prefix when accessing keys. Isn't that the case? I.e. to rephrase the question: Is prefix implicit on one of the sides?

Reply:
No but

`(prefix,key)`
`("storage/magic", "birdy")`
`("storage", "magic/birdy")`
become equiv in the off-chain storage overlay, and as such you get into not very nice to debug issues.

Reply:
Will be addressed in a separate PR.